Submitted By: Gregory Davis <gregdavis AT ieee DOT org>
Date: 2003-10-31
Initial Package Version: 2.6.0-test8
Origin: Gregory Davis
Description: Fixes broken generation of linux kernel (2.6.0-test8) documentation.
--- linux-2.6.0-test8-vanilla/Documentation/DocBook/Makefile	2003-10-17 17:43:10.000000000 -0400
+++ linux-2.6.0-test8-working/Documentation/DocBook/Makefile	2003-10-27 13:14:14.000000000 -0500
@@ -14,6 +14,9 @@
 	    kernel-api.sgml journal-api.sgml lsm.sgml usb.sgml \
 	    gadget.sgml
 
+DCL = /usr/share/sgml/docbook/dsssl-stylesheets-1.78/dtds/decls/docbook.dcl
+DSLDIR = /usr/share/sgml/docbook/dsssl-stylesheets-1.78
+
 ###
 # The build process is as follows (targets):
 #              (sgmldocs)
@@ -104,35 +107,39 @@
 
 ###
 # Rules to generate postscript, PDF and HTML
-# db2html creates a directory. Generate a html file used for timestamp
+# docbook2html creates a directory. Generate a html file used for timestamp
 
-quiet_cmd_db2ps = DB2PS   $@
-      cmd_db2ps = db2ps -o $(dir $@) $<
+quiet_cmd_docbook2ps = DB2PS   $@
+      cmd_docbook2ps = docbook2ps -d $(DSLDIR)/print/docbook.dsl -l $(DCL) \
+      -o $(dir $@) $<
 %.ps : %.sgml
-	@(which db2ps > /dev/null 2>&1) || \
+	@(which docbook2ps > /dev/null 2>&1) || \
 	 (echo "*** You need to install DocBook stylesheets ***"; \
 	  exit 1)
-	$(call cmd,db2ps)
+	$(call cmd,docbook2ps)
 
-quiet_cmd_db2pdf = DB2PDF  $@
-      cmd_db2pdf = db2pdf -o $(dir $@) $<
+quiet_cmd_docbook2pdf = DB2PDF  $@
+      cmd_docbook2pdf = docbook2pdf -d $(DSLDIR)/print/docbook.dsl -l $(DCL) \
+      -o $(dir $@) $<
+      
 %.pdf : %.sgml
-	@(which db2pdf > /dev/null 2>&1) || \
+	@(which docbook2pdf > /dev/null 2>&1) || \
 	 (echo "*** You need to install DocBook stylesheets ***"; \
 	  exit 1)
-	$(call cmd,db2pdf)
+	$(call cmd,docbook2pdf)
 
-quiet_cmd_db2html = DB2HTML $@
-      cmd_db2html = db2html -o $(patsubst %.html,%,$@) $< &&		      \
-		echo '<a HREF="$(patsubst %.html,%,$(notdir $@))/book1.html"> \
+quiet_cmd_docbook2html = DB2HTML $@
+      cmd_docbook2html = docbook2html -d $(DSLDIR)/html/docbook.dsl -l $(DCL) \
+      -o $(patsubst %.html,%,$@) $< &&		                          \
+		echo '<a HREF="$(patsubst %.html,%,$(notdir $@))/book1.htm"> \
          Goto $(patsubst %.html,%,$(notdir $@))</a><p>' > $@
 
 %.html:	%.sgml
-	@(which db2html > /dev/null 2>&1) || \
+	@(which docbook2html > /dev/null 2>&1) || \
 	 (echo "*** You need to install DocBook stylesheets ***"; \
 	  exit 1)
 	@rm -rf $@ $(patsubst %.html,%,$@)
-	$(call cmd,db2html)
+	$(call cmd,docbook2html)
 	@if [ ! -z "$(PNG-$(basename $(notdir $@)))" ]; then \
             cp $(PNG-$(basename $(notdir $@))) $(patsubst %.html,%,$@); fi
 
